app/main/views/services.py

```diff
@@ -159,8 +159,10 @@ def update_service(service_id):
             abort(400, "Invalid value for 'copiedToFollowingFramework' supplied")
         service.copied_to_following_framework = update['copiedToFollowingFramework']
 
-    if 'supplierId' in update:
+    if 'supplierId' in update and update['supplierId'] != service.supplier_id:
```

Comment:
Does the endpoint allow you to assign a string and it'll automagically convert it to an int? If so you'd need to wrap the left hand side in `int()` too to be sure.

Reply:
Good shout.